tests/DocumentationExamplesTest.php

```diff
@@ -1180,6 +1184,223 @@ public function testIndex_example_2()
+                echo "Error during commit ...\n";
+                throw $error;
+            }
+        }
+    }
+    // End Transactions Intro Example 1
+
+    public function testTransactions_intro_example_1()
+    {
+        $hrDb = new Database($this->manager, "hr");
+        $reportingDb = new Database($this->manager, "reporting");
+
+        $hrDb->dropCollection('employees');
+        $reportingDb->dropCollection('events');
+
+        /* Collections need to be created before a transaction starts */
+        $hrDb->createCollection('employees');
+        $reportingDb->createCollection('events');
+
+        $session = $this->manager->startSession();
+
+        $this->updateEmployeeInfo1($this->manager, $session);
+    }
+
+    // Start Transactions Retry Example 1
+    private function runTransactionWithRetry1(callable $txnFunc, Manager $manager, Session $session)
+    {
+        while (true) {
+            try {
+                $txnFunc($manager, $session);  // performs transaction
+                break;
+            } catch (\MongoDB\Driver\Exception\CommandException $error) {
+                $resultDoc = $error->getResultDocument();
+                echo "Transaction aborted. Caught exception during transaction.\n";
+
+                // If transient error, retry the whole transaction
+                if (isset($resultDoc->errorLabels) && in_array('TransientTransactionError', $resultDoc->errorLabels)) {
+                    echo "TransientTransactionError, retrying transaction ...\n";
+                    continue;
+                } else {
+                    throw $error;
+                }
+            } catch (\MongoDB\Driver\Exception\Exception $error) {
+                throw $error;
+            }
+        }
+    }
+    // End Transactions Retry Example 1
+
+    // Start Transactions Retry Example 2
+    private function commitWithRetry2(Manager $manager, Session $session)
+    {
+        while (true) {
+            try {
+                $session->commitTransaction();
+                echo "Transaction committed.\n";
+                break;
+            } catch (\MongoDB\Driver\Exception\CommandException $error) {
+                $resultDoc = $error->getResultDocument();
+
+                if (isset($resultDoc->errorLabels) && in_array('UnknownTransactionCommitResult', $resultDoc->errorLabels)) {
+                    echo "UnknownTransactionCommitResult, retrying commit operation ...\n";
+                    continue;
+                } else {
+                    echo "Error during commit ...\n";
+                    throw $error;
+                }
+            } catch (\MongoDB\Driver\Exception\Exception $error) {
+                echo "Error during commit ...\n";
+                throw $error;
+            }
+        }
+    }
+    // End Transactions Retry Example 2
+
+    // Start Transactions Retry Example 3
+    private function runTransactionWithRetry3(callable $txnFunc, Manager $manager, Session $session)
```

Comment:
Second argument can be `\MongoDB\Database $db`.

Reply:
Done